tests/mechanic/launcher_test.py

```diff
@@ -141,6 +141,21 @@ def test_launches_cluster_without_post_launch_handler(self):
         self.assertEqual([{"host": "10.0.0.10", "port":9200}, {"host": "10.0.0.11", "port":9200}], cluster.hosts)
         self.assertIsNotNone(cluster.telemetry)
 
+    def test_launches_cluster_with_telemetry_client_timeout_enabled(self):
+        cfg = config.Config()
+        cfg.add(config.Scope.application, "client", "hosts", self.test_host)
+        cfg.add(config.Scope.application, "client", "options", self.client_options)
+        cfg.add(config.Scope.application, "mechanic", "telemetry.devices", [])
+        cfg.add(config.Scope.application, "mechanic", "telemetry.params", {})
+
+        cluster_launcher = launcher.ClusterLauncher(cfg, MockMetricsStore(), client_factory_class=MockClientFactory)
+        cluster = cluster_launcher.start()
+
+        retry_timeout_setting = {"retry-on-timeout": True}
+
+        for telemetry_device in cluster.telemetry.devices:
+            self.assertLessEqual(retry_timeout_setting.items(),telemetry_device.client.client_options.items())
```

Comment:
We could use `assertDictEqual` here to check whether the client dictionary contains only expected options?

Reply:
My thought here is that this test cares about ensuring that the specific option is set and we won't need to update it, if, for whatever reason, other client option settings, specified in the Class Attributes, change in the future. I don't mind changing it to what you are suggesting though.